functions/go/bind/pkg/meta/namespaceref.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package meta
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type namespaceRef struct {
+	parentObject *fn.SubObject
+
+	name      string
+	fieldPath []string
+}
+
+func buildMetadataNamespaceReference(parentObject *fn.KubeObject) (Ref, error) {
+	namespace := parentObject.GetMap("metadata").GetString("namespace")
+	if namespace == "" {
+		return nil, fmt.Errorf("expected namespace to be set")
+	}
+	return &namespaceRef{
+		name:         namespace,
+		parentObject: &parentObject.SubObject,
+		fieldPath:    []string{"metadata", "namespace"},
+	}, nil
+}
+
+// buildRefNamespaceReference returns a Ref represented the namespace in a reference
+func buildRefNamespaceReference(ref *fn.SubObject, fieldPath ...string) (Ref, error) {
```

Comment:
I have a proposal for the next step of this buildRefNamespaceReference I'd like to hear your opinions:
I want to have this buildRef function to not request a specific fieldPath, but can discover the "namespace" reference it self, basically it can walk through all the descendant SubObjects and look for "namespace" field (we might want to set a condition that the referred resource should exist in the input ResourceLis). For example,
 "ConfigControllerInstance" has "spec.managementConfig.standardManagementConfig.networkRef.namespace` which needs to have the network "namespace"  updated, the function discover the path and update the "namespace" as well.

Reply:
Ah, never mind. I just realized this PR already does this part (`VisitRefs` is cool)!